### PR TITLE
Finish Build Exercise Screen

### PR DESCRIPTION
**Pull request by ArielWangX**
Create component `ScaffoldWithTopBar.kt `
Add TopBar to the exercise screen.
Add ConstrainLayout to CardList and Button.

Exercise Screen Recording:

https://user-images.githubusercontent.com/109891752/194221007-04a4e150-8cae-4473-bb40-964ff5098822.mp4

### CODE REVIEW

**Comment by fredliao123** [comment]
It would be better to add a screenshot of the page to this PR

File: app/src/main/java/com/arielwang/workoutlogger/features/component/TopBarScaffold.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            TopAppBar(
+                title = {
+                    Text(text = stringResource(id = title))
+                },
+                navigationIcon = {
+                    IconButton(onClick = { onAction() }) {
+                        Icon(Icons.Filled.ArrowBack, contentDescription = stringResource(id = iconContentDescription))
+                    }
+                },
+                backgroundColor = MaterialTheme.colors.onPrimary,
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/component/TopBarScaffold.kt:35]: I know the design uses onPrimary on the TopAppBar. But would it makes more sense to follow the [material design guideline](https://material.io/design/color/the-color-system.html#color-theme-creation) and use the primary color

File: app/src/main/java/com/arielwang/workoutlogger/features/component/TopBarScaffold.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                contentColor = MaterialTheme.colors.secondary,
+                elevation = 10.dp
+            )
+        },
+
+        content = {
+            Column(
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/component/TopBarScaffold.kt]: In the ExeciseScreen, the content is a constraint layout, this column doesn't have any effect. For a commonly used UI component, this shouldn't preassume the content layout

File: app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    onAction: (ExerciseView.Action) -> Unit
+) {
+    ConstraintLayout(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = 16.dp)
+    ) {
+        val (button, cardListSection) = createRefs()
+
+        ExerciseCardList(
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt:49]: I like how these composables have their own functions 👍

File: app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+@Composable
+fun ExerciseCardList(
+    uiState: ExerciseView.State,
+    onAction: (ExerciseView.Action) -> Unit,
+    cardListConstrainLayout: Modifier
+) {
+    LazyColumn(
+        modifier = cardListConstrainLayout
+    ) {
+        item {Spacer(modifier = Modifier.height(30.dp))}
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt]: We always want to use multiple of 8 in these dimensions like 16.dp, 24.dp, 32.dp
https://uxplanet.org/everything-you-should-know-about-8-point-grid-system-in-ux-design-b69cb945b18d

File: app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+        uiState.cardList.forEach {
+            item { ExerciseCard(text = it.text, isSelected = it.isSelected, onAction = onAction) }
+        }
+
+        item {Spacer(modifier = Modifier.height(20.dp))}
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt]: Same here

File: app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+@Composable
+fun ExerciseCard(
+    text: String,
+    isSelected: Boolean,
+    onAction: (ExerciseView.Action) -> Unit
+) {
+    Card(
+        border = BorderStroke(
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt:95]: 👍

File: app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                    cardList = viewState.cardList.map {
+                        if (it.text == action.text) {
+                            return@map it.copy(isSelected = !it.isSelected)
+                        } else {
+                            return@map it
+                        }
+                    }
+                )
+
+//  different approach
```
Comment (fredliao123) [inline app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseViewModel.kt]: We should remove these in the PR

**Review by fredliao123 — COMMENTED** [review]
Really good work. Just have a few comments

File: app/src/main/java/com/arielwang/workoutlogger/features/component/TopBarScaffold.kt (hunk 1 above)
Comment (ArielWangX) [reply app/src/main/java/com/arielwang/workoutlogger/features/component/TopBarScaffold.kt:35]: thank you so much for introducing material design guideline to me. It is very helpful.

File: app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt (hunk 4 above)
Comment (ArielWangX) [reply app/src/main/java/com/arielwang/workoutlogger/features/exercise/ui/screen/ExerciseScreen.kt]: Thank you so much for pointing this out.

**Review by fredliao123 — APPROVED** [review]
Great work

**Pull request merged**